uilib/src/components/Loader/index.tsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import { Svg, Circle, G, Rect } from "react-native-svg";
+import { useTheme } from "styled-components/native";
+import { TouchableOpacity } from "react-native";
+
+type Props = {
+  // float number between 0 and 1
+  progress?: number;
+
+  // function triggered when pressing the loader
+  onPress: () => void;
+
+  // Display the square in the middle of the loader
+  displayCancelIcon?: boolean;
+};
+
+const R = 25;
+
+const ProgressLoader = ({
+  progress = 0,
+  onPress,
+  displayCancelIcon = true,
+}: Props): React.ReactElement => {
+  const { colors } = useTheme();
+  const backgroundColor = colors.palette.primary.backgroundLight;
+  const progressColor = colors.palette.primary.dark;
+
+  const strokeWidth = 4;
+  const normalizedRadius = R - strokeWidth / 2;
+  const circumference = normalizedRadius * 2 * Math.PI;
+  const strokeDashoffset = circumference - progress * circumference;
+  return (
+    <TouchableOpacity onPress={onPress}>
```

Comment:
Wondering if the loader component should only be touchable if it has an `onPress` prop (and make the prop optional btw) 🤔. I think most of the time a loader is not meant to be interactive?

Reply:
Fixed, thanks :+1: